packages/core/contracts/financial-templates/perpetual-multiparty/PerpetualCreator.sol

```diff
@@ -65,7 +70,25 @@ contract PerpetualCreator is ContractCreator, Testable, Lockable {
      * @return address of the deployed contract.
      */
     function createPerpetual(Params memory params) public nonReentrant() returns (address) {
-        address derivative = PerpetualLib.deploy(_convertParams(params));
+        // Create a new synthetic token using the params.
+        require(bytes(params.syntheticName).length != 0, "Missing synthetic name");
+        require(bytes(params.syntheticSymbol).length != 0, "Missing synthetic symbol");
+        TokenFactoryExclusiveMinter tf = TokenFactoryExclusiveMinter(tokenFactoryAddress);
+        ExpandedIERC20ExclusiveMinter tokenCurrency = tf.createToken(
+            params.syntheticName,
+            params.syntheticSymbol,
+            IERC20Standard(params.collateralAddress).decimals()
```

Comment:
Should we throw a `try-catch` around this and default to 18?

Reply:
I presume this would account for currencies without a decimals method. So what if a currency doesn't have decimals but it uses non 18 precision? I guess we can address this by changing the precision of the UMIP, or maybe we can allow it to be passed into the creator

Reply:
added a try-catch around the external contract call to `decimals()`

Reply:
try catch usage details here: https://solidity.readthedocs.io/en/latest/control-structures.html#try-catch